### DISCUSSION TITLE
Two-way binding between components using Relm4

### DISCUSSION THREAD

**Discussion by chriskilding** [Relm4/Relm4#493] [Q&A]
I was experimenting with having multiple components that share a two-way binding, and am wondering if this is possible in Relm4. I saw that bindings were added in 0.6.0, but I'm not yet sure whether they would be the solution.

For example, say that your app has 2 `gtk::ScrolledWindow`s where you want to share the horizontal scroll position / adjustment. In other words, scrolling on either of them should scroll the other to the same position. (Assume for these purposes that we have ensured that the content of both `ScrolledWindow`s is the same width.)

How would you achieve that with Relm4?

**Comment by AaronErhardt** [comment]
I think there are several solutions. You could use a `thread_local` static property binding and bind both scroll positions to it.

Also, you could use `MessageBroker`s to send messages between both components (or use `forward` if they have the same parent). This is rather inefficient for frequent events like scroll position changes though.
